seqr/views/utils/terra_api_utils.py

```diff
@@ -0,0 +1,177 @@
+"""
+This module provides python bindings for the AnVIL Terra API.
+"""
+
+import json
+import logging
+
+from urllib.parse import urljoin
+
+from google.auth.transport.requests import AuthorizedSession
+from google.oauth2 import service_account
+
+from settings import SEQR_VERSION, TERRA_API_CONFIG
+
+SEQR_USER_AGENT = "seqr/" + SEQR_VERSION
+
+scopes = ['https://www.googleapis.com/auth/userinfo.profile',
+          'https://www.googleapis.com/auth/userinfo.email',
+          'https://www.googleapis.com/auth/cloud-billing',
+          'openid']
+
+logger = logging.getLogger(__name__)
+
+
+class TerraAPIException(Exception):
+    """For exceptions happen in Terra API calls
+    """
+    pass
+
+
+def _seqr_agent_header(headers=None):
+    """ Return request headers for Terra API message.
+        Inserts seqr/version as the User-Agent.
+    Args:
+        headers (dict): Include additional headers as key-value pairs
+    """
+    seqr_headers = {"User-Agent": SEQR_USER_AGENT}
+    if headers is not None:
+        seqr_headers.update(headers)
+    return seqr_headers
+
+
+class AnvilSession:
```

Comment:
it seems like this would be much neater if this class inherited from `AuthorizedSession`, instead of having an instance of that class inside it

Reply:
It could be. Let me think more about it.